web/src/main/kotlin/no/nav/su/se/bakover/web/routes/me/MeRoutes.kt

```diff
@@ -0,0 +1,58 @@
+package no.nav.su.se.bakover.web.routes.me
+
+import arrow.core.Either
+import io.ktor.application.call
+import io.ktor.auth.authentication
+import io.ktor.http.HttpStatusCode
+import io.ktor.response.respond
+import io.ktor.routing.Route
+import io.ktor.routing.get
+import no.nav.su.se.bakover.common.Config
+import no.nav.su.se.bakover.common.filterMap
+import no.nav.su.se.bakover.common.serialize
+import no.nav.su.se.bakover.web.features.suUserContext
+import no.nav.su.se.bakover.web.getGroupsFromJWT
+
+enum class Rolle(val type: String) {
+    Attestant("ATTESTANT"),
+    Saksbehandler("SAKSBEHANDLER"),
+    Veileder("VEILEDER")
+}
+
+data class UserData(
+    val navIdent: String,
+    val roller: List<Rolle>
+)
+
+@OptIn(io.ktor.locations.KtorExperimentalLocationsAPI::class)
+internal fun Route.meRoutes() {
+    get("/me") {
+        val user = call.suUserContext.user!!
+        val prince = call.authentication.principal
+        val roller = getGroupsFromJWT(prince).filterMap {
+            when (it) {
```

Comment:
Jeg tolker `filterMap` som at den kanskje "akkumulerer" resultatet av alt som evaluerer til true? Isåfall er vel flere roller håndtert.

Reply:
Ja og ja. Ønsker å lage en liste med alle "kjente" grupper. Er kanskje litt forvirrende at vi kaller det `roller` videre. Her er vi litt i klinsj mellom hva det kalles i AD (grupper) og hva vi og fag bruker i dagligtalen (rolle). Burde vi velge ett begrep for å være konsistente i koden, eller skal vi fortsette på dette sporet og kalle det en implementasjonsdetalj at det kalles grupper i AD?

Reply:
Jeg synes vi skal bruke domeneordet i domenet, mens akkurat i selve AD-laget, kan vi kalle det for implementasjonsdetaljen. Også har det laget ansvaret for mappingen.

Reply:
Mener du da sånn det har blitt gjort her?

Reply:
Jepp. Kunne ha flyttet enum class Rolle til domene og lagt på litt mer kjøtt i navnet; Brukerrolle eller noe.